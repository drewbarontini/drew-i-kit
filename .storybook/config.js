import { configure, addDecorator } from '@storybook/react';
import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import 'normalize.css';

import { theme } from '../config';

import GlobalStyles from '../components/GlobalStyles';

const req = require.context('../components/', true, /.story.js$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

function withStyledComponents(story) {
  return (
    <ThemeProvider theme={theme}>
      <Fragment>
        {story()}
        <GlobalStyles />
      </Fragment>
    </ThemeProvider>
  );
}

addDecorator(withStyledComponents);
configure(loadStories, module);
