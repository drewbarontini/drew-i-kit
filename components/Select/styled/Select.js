import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp } from '../../../lib/utils';

const Select = styled.select`
  ${setProp({
    prop: 'size',
    cssProp: 'width',
    fallback: '200px',
  })}
`;

Select.propTypes = {
  size: PropTypes.string,
};

Select.displayName = 'StyledSelect';

export default Select;
