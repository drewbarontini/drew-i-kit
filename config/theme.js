const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
