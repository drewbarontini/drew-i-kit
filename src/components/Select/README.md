# Select

### Usage

```jsx
import { Select } from 'drew-i-kit';

<Select
  name="select"
  options={[{ value: 'one', label: 'One' }, { value: 'two', label: 'Two' }]}
/>
```

```jsx
import { Select } from 'drew-i-kit';

<Select
  name="select"
  options={[{ value: 'one', label: 'One' }, { value: 'two', label: 'Two' }]}
  placeholder="Select an option..."
/>
```

### Props

- `name` (**string**)
- `options` (**Object**)
- `placeholder` (**string**, optional)

_Also includes all default props for `<select>` elements._