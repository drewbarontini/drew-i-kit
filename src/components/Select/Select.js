import React from 'react';
import PropTypes from 'prop-types';

import StyledSelect from './styled/Select';

const Select = ({ name, placeholder, options, ...rest }) => (
  <StyledSelect name={name} id={name} {...rest}>
    <option value="">{placeholder}</option>
    {options.map((option, index) => (
      <option key={index} value={option.value}>
        {option.label}
      </option>
    ))}
  </StyledSelect>
);

Select.propTypes = {
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  placeholder: PropTypes.string,
};

Select.defaultProps = {
  placeholder: 'Select...',
};

Select.displayName = 'Select';

export default Select;
