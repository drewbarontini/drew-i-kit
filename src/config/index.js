export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
