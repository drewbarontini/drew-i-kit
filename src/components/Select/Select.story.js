import React from 'react';
import { storiesOf } from '@storybook/react';

import Select from './';

storiesOf('Select', module).add('default', () => (
  <Select
    name="select"
    options={[{ value: 'foo', label: 'Foo' }, { value: 'bar', label: 'Bar' }]}
  />
));
