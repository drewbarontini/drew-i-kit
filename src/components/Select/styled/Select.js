import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp } from '../../../lib/props';
import { responsive } from '../../../lib/responsive';

const Select = styled.select`
  ${setProp({
    prop: 'size',
    cssProp: 'width',
    fallback: '200px',
  })};

  ${responsive};
`;

Select.propTypes = {
  size: PropTypes.string,
};

Select.displayName = 'StyledSelect';

export default Select;
