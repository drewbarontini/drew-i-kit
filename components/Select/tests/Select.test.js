import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import Select from '../';

describe('<Select />', () => {
  it('should render properly', () => {
    const { container, debug } = render(
      <Select
        theme={theme}
        name="select"
        options={[
          { value: 'foo', label: 'Foo' },
          { value: 'bar', label: 'Bar' },
        ]}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.tagName).toEqual('SELECT');
    expect(container.firstChild).toHaveAttribute('id');
    expect(container.firstChild.id).toEqual('select');
    expect(container.firstChild).toHaveAttribute('name');
    expect(container.firstChild.name).toEqual('select');
    expect(container.firstChild.childNodes.length).toBe(3);
  });
});
