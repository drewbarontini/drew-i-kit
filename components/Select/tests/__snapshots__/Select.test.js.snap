// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render properly 1`] = `
.c0 {
  width: 200px;
}

<select
  class="c0"
  id="select"
  name="select"
>
  <option
    value=""
  >
    Select...
  </option>
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;
